erase 'latest' tag before releasing again

israpps opened the pull request:
This must be done because the release step used for this workflow doesn't cleanup previous release.

This has 2 concecuences:

1. release tag is tied to the 1st commit released by this workflow step
2. since artifacts have sha inside filename the artifact will vary, stacking over time inside the same release tag... instead of only providing artifact generated by latest commit

israpps committed:
erase 'latest' tag before releasing again

This must be done because the release step used for this workflow doesn't cleanup previous release.

This has 2 concecuences:

1) release tag is tied to the 1st commit released by this workflow step
2) since artifacts have sha inside filename the artifact will vary, stacking over time inside the same release tag...

AKuHAK commented:
Not quite sure why this needed to be done.

fjtrujy commented:
Me neither

AKuHAK commented:
@fjtrujy If I remember correctly - `latest` tag is used for proper action dispatching. It is not something that is "redundant".

fjtrujy commented:
If I'm right "latest" is like "snapshot" is like the latest binary generated for the latest master status.

This latest tag is "updated" everytime that a new commit is done to `master` branch

israpps commented:
> If I'm right "latest" is like "snapshot" is like the latest binary generated for the latest master status.
> 
> This latest tag is "updated" everytime that a new commit is done to `master` branch

Look at latest release..

You have binaries of 3 different releases

fjtrujy commented:
> > If I'm right "latest" is like "snapshot" is like the latest binary generated for the latest master status.
> > This latest tag is "updated" everytime that a new commit is done to `master` branch
> 
> Look at latest release..
> 
> You have binaries of 3 different releases

Now I see it. Thanks

israpps committed:
Clean release before matrix to avoid issues

israpps committed:
Move the condition to job

That way workflows triggered by pull requests won't have to set up a job that won't run anything

israpps commented:
@AKuHAK don't forget of this PR
![Captura de pantalla 2022-05-11 143111](https://user-images.githubusercontent.com/57065102/167911131-009d985a-7c60-4ad6-a420-9ad2c77b240a.png)

uyjulian commented:
I undid the merge because it broke CI: https://github.com/ps2dev/ps2link/actions/runs/2308706185

israpps commented:
sorry, i saw it
let me fix this
